### PR TITLE
Skip subsampling if batch size is less than n_samples_subsample

### PR DESCRIPTION
**Pull request by kjaisingh**
This PR addresses Issue #706.

### Description
Includes a check that `n_samples_subsample` is less than the batch size - i.e. `length(samples)`. If this is the case, the subsampling task `RandomSubsampleStringArray` is skipped.

### Testing
[This Terra job](https://app.terra.bio/#workspaces/Talkowski_training/kj-issue-701-gatk-sv-pipeline-1kgp/job_history/27b982d8-fd24-4bbc-862c-1a47cf683fab) shows an example run of the pipeline for the 1KGP cohort with this change. It does not call `RandomSubsampleStringArray` as the number of samples (5) is less than the default value for `n_samples_subsample` (100). 

### Pre-Merge Changes Required
Remove automated Dockstore image sync for `TrainGCNV.wdl` in development branch.

### CODE REVIEW

File: wdl/TrainGCNV.wdl [hunk 1]
```diff
@@ -102,6 +102,8 @@ workflow TrainGCNV {
     RuntimeAttr? runtime_attr_explode
   }
 
+  Boolean perform_subsampling = defined(n_samples_subsample) && (select_first([n_samples_subsample]) < length(samples))
```
Comment (epiercehoffman) [inline wdl/TrainGCNV.wdl:105]: I would probably put this right above the check for whether to run RandomSubsampleStringArray since that's where it's used

**Review by epiercehoffman — APPROVED** [review]
Looks good, thanks! Make sure to revert the change to the dockstore.yml before merging. One small style comment

File: wdl/TrainGCNV.wdl (hunk 1 above)
Comment (kjaisingh) [reply wdl/TrainGCNV.wdl]: Thanks. Just pushed a new commit that integrates this new variable into the existing if statement that checks whether to run `RandomSubsampleStringArray` - let me know if this is better, or if I should pull it out into a new variable like I'd done here.

Comment (epiercehoffman) [reply wdl/TrainGCNV.wdl]: I don't have strong feelings either way so this looks good to me!

**Review by epiercehoffman — CHANGES_REQUESTED** [review]
Can you also update the docs to reflect this change? Sorry I missed this on my first review

**Comment by kjaisingh** [comment]
> Can you also update the docs to reflect this change? Sorry I missed this on my first review

Updated this in my latest commit.

File: README.md [hunk 2]
```diff
@@ -215,7 +215,7 @@ For larger cohorts, samples should be split up into batches of about 100-500 sam
...
+Both the cohort and single-sample modes use the [GATK-gCNV](https://gatk.broadinstitute.org/hc/en-us/articles/360035531152) depth calling pipeline, which requires a [trained model](#gcnv-training) as input. The samples used for training should be technically homogeneous and similar to the samples to be processed (i.e. same sample type, library prep protocol, sequencer, sequencing center, etc.). The samples to be processed may comprise all or a subset of the training set. For small, relatively homogenous cohorts, a single gCNV model is usually sufficient. If a cohort contains multiple data sources, we recommend training a separate model for each [batch](#batching) or group of batches with similar dosage score (WGD). The model may be trained on all or a subset of the samples to which it will be applied; a reasonable default is 100 randomly-selected samples from the batch. The random selection can be done as part of the workflow by specifying a number of samples to the `n_samples_subsample` input parameter in `/wdl/TrainGCNV.wdl`, though this step is skipped if the value passed in via `n_samples_subsample` is smaller than the batch size.
```
Comment (epiercehoffman) [inline README.md:218]: Thanks for being so quick with this update! I should be more specific though. I think this detail is probably not necessary to mention in this section as it's more of an overview. This is the part of the documentation that's out of sync with this change: https://github.com/broadinstitute/gatk-sv/blob/main/inputs/templates/terra_workspaces/cohort_mode/cohort_mode_workspace_dashboard.md.tmpl#L169

Comment (kjaisingh) [reply README.md]: Got it, thanks for flagging. Updated accordingly.

File: inputs/templates/terra_workspaces/cohort_mode/cohort_mode_workspace_dashboard.md.tmpl [hunk 3]
```diff
@@ -166,7 +166,7 @@ Read the full EvidenceQC documentation [here](https://github.com/broadinstitute/
 
 Read the full TrainGCNV documentation [here](https://github.com/broadinstitute/gatk-sv#gcnv-training-1).
 * Before running this workflow, create the batches (~100-500 samples) you will use for the rest of the pipeline based on sample coverage, WGD score (from `02-EvidenceQC`), and PCR status. These will likely not be the same as the batches you used for `02-EvidenceQC`.
-* By default, `03-TrainGCNV` is configured to be run once per `sample_set` on 100 randomly-chosen samples from that set to create a gCNV model for each batch. If your `sample_set` contains fewer than 100 samples (not recommended), you will need to edit the `n_samples_subsample` parameter to be less than or equal to the number of samples.
+* By default, `03-TrainGCNV` is configured to be run once per `sample_set` on 100 randomly-chosen samples from that set to create a gCNV model for each batch. If your `sample_set` contains fewer than 100 samples (not recommended), this step is skipped - however, you can edit the `n_samples_subsample` parameter to be less than or equal to the number of samples for this to run.
```
[the remark this answers is not in the document]
Comment (kjaisingh) [reply inputs/templates/terra_workspaces/cohort_mode/cohort_mode_workspace_dashboard.md.tmpl]: I like the edit over just removing the line - providing an explicit reference to `n_samples_subsample` enables users to know how they can run random subsampling on a smaller sample size if they have < 100 samples (even though this is not ideal).

**Review by epiercehoffman — APPROVED** [review]
I think this is all set now! Thanks for bearing with me on the docs details

**Pull request merged**